### ISSUE TITLE
i'm not sure about this comment 

### ISSUE THREAD

**Issue by alexlib** [yosefm/pbi#3]
https://github.com/yosefm/pbi/blob/6dd7b54881fbfa15e305c334da56e278a801ac16/ptv/sequence.py#L78

@yosefm please can you explain me what does it mean "only after they've been corrected"? does it mean that `correspondences` might change detections? maybe only the `sorted` has to be written?

**Comment by yosefm** [comment]
the change is in bookkeeping. correspondence information is added to the
targets.

בתאריך יום ו׳, 30 במרץ 2018, 08:52, מאת Alex Liberzon ‏<
notifications@github.com>:

**Comment by alexlib** [comment]
I remember the book keeping story, thanks. but I do not understand how it's done. I mean that we have already `detections` and `corrected` arrays, that we send to `correspondences`:
```
sets, corresp, _ = correspondences(
            detections, corrected, self._cals, self._vpar, self._cpar)
```
and after that we write `detections`:
```
# Save targets only after they've been modified:
        for cix in xrange(len(self._cams)):
            detections[cix].write(
                seq['targets_template'].format(cam=cix + 1), frame)
```

does it mean that `detections` have been modified within `correspondences`?

**Comment by yosefm** [comment]
"corrected" is just the flat-image coordinates (was called `fix` in 3dptv).
`detections` holds the 3d particle number after one has been assigned to it
- that is the change that happens in correspondences.

**Comment by alexlib** [comment]
sorry, but I still do not understand. 

Here: 

```
sets, corresp, _ = correspondences(
            detections, corrected, self._cals, self._vpar, self._cpar)
```
we send `detections` as an input to `correspondences`. Obviously even if it is changed inside `correspondences.pyx` - it is not returned as an output. What do I miss here?

**Comment by yosefm** [comment]
Some functions have side-effects. They treat some of their inputs as
outputs as well. It's not so clean and nice, but sometimes it's necesary.
In this case, if you have a lot of targets, it's a lot of burden to copy
them all over just for the one changed property. At least I made it cleaner
than it was before...

I also documented it in correspondences.pyx:
"""
img_pts - a list of c := len(cals), containing TargetArray objects, each
+ with the target coordinates of n detections in the respective image.
+ The target arrays are clobbered: returned arrays have the tnr property
+ set. the pnr property should be set to the target index in its array.
"""
etc. There's a pretty long docstring.

I hope this helps to clarify.
